x/compliancetest/types/message_add_testing_result_test.go

```diff
@@ -1,42 +1,135 @@
 package types
 
-/* TODO issue 99
 import (
 	"testing"
+	"time"
 
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/stretchr/testify/require"
+	testconstants "github.com/zigbee-alliance/distributed-compliance-ledger/integration_tests/constants"
 	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/sample"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/utils/validator"
 )
 
 func TestMsgAddTestingResult_ValidateBasic(t *testing.T) {
-	tests := []struct {
+	negative_tests := []struct {
 		name string
 		msg  MsgAddTestingResult
 		err  error
 	}{
 		{
 			name: "invalid address",
 			msg: MsgAddTestingResult{
-				Signer: "invalid_address",
+				Signer:                "invalid_address",
+				SoftwareVersionString: testconstants.SoftwareVersionString,
```

Comment:
It's better to set `SoftwareVersion` everywhere explicitly, and add one explicit positive test where SoftwareVersion=0.